lib/spack/llnl/util/filesystem.py

```diff
@@ -196,10 +196,11 @@ def change_sed_delimiter(old_delim, new_delim, *filenames):
 
 def set_install_permissions(path):
     """Set appropriate permissions on the installed file."""
-    if os.path.isdir(path):
-        os.chmod(path, 0o755)
-    else:
-        os.chmod(path, 0o644)
+    if not os.path.islink(path):
+        if os.path.isdir(path):
+            os.chmod(path, 0o755)
+        if os.path.isfile(path):
+            os.chmod(path, 0o644)
 
 
 def copy_mode(src, dest):
```

Comment:
I think it would be worth adding a docstring mentioning that this has no affect on a symlink (or the target of a symlink).